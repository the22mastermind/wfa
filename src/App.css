:root {
  --customBlue: #072942;
  --customYellow: #E69646;
  --pureWhite: #FFFFFF;
}
* {
  box-sizing: border-box;
  outline: 0;
}
body {
  background: url('./assets/img/background.svg') no-repeat center center fixed;
  -wekit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  font-family: 'Open Sans', sans-serif;
  color: var(--customBlue);
}
nav {
  padding: 10px;
}
.brand {
  text-decoration: none;
  color: var(--customBlue);
}
.nav-links {
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style-type: none;
  padding: 0;
}
.nav-links > li > a {
  text-decoration: none;
}
/* content */
.container {
  display: flex;
  height: 100%;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}
.form-wrapper {
  margin-top: 20px;
  background-color: rgba(256, 256, 256, 0.8);
  border-radius: 30px;
}
.form-title {
  font-weight: 700;
  text-align: center;
  margin-bottom: 20px;
  color: var(--customBlue);
}
.names {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}
.input {
  background-color: transparent;
  border: none;
  border-bottom: 1px solid var(--customBlue);
  height: 34px;
  width: 100%;
  font-size: 10pt;
  color: var(--customBlue);
}
::placeholder {
  color: var(--customBlue);
  opacity: 1;
}
.submit-btn-wrapper {
  text-align: center;
}
.submit-btn {
  background-color: var(--customYellow);
  color: var(--customBlue);
  border: 1px solid var(--customYellow);
  border-radius: 10px;
}
.alt-wrapper {
  font-size: 10pt;
  text-align: center;
}
.alt-text {
  margin-right: 5px;
}
.alt-wrapper > .link {
  font-weight: 700;
}
a,
a:hover,
a:focus {
  text-decoration: none;
}
.link {
  color: var(--customYellow);
}
html,
body,
#root {
  height: 100%;
  margin: 0;
}
.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 10px;
}
/* End of content */

@media (max-width: 575.98px) {
  /* Navbar */
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .nav-links {
    flex-direction: row;
  }
  .brand {
    font-size: 12pt;
  }
  .nav-links > li > .link {
    margin-right: 20px;
    color: var(--customBlue);
  }
  .nav-links > li > a {
    font-size: 10pt;
  }
  .small-btn {
    padding: 10px;
    background-color: var(--customBlue);
    border-radius: 10px;
    color: var(--pureWhite);
  }
  /* End of navbar */
  /* Content */
  .landing-page-container {
    width: 100%;
    bottom: 20px;
    position: absolute;
  }
  .landing-page-content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 250px;
    margin-bottom: 20px;
  }
  .landing-page-text {
    width: 80%;
    height: 80%;
    margin-bottom: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .landing-page-title {
    color: var(--pureWhite);
    text-align: center;
    font-size: 18pt;
  }
  .cta-btn {
    padding: 8px 18px;
    background-color: var(--customYellow);
    color: var(--customBlue);
    border: 1px solid var(--customYellow);
    border-radius: 10px;
    margin-top: 20px;
    text-decoration: none;
    font-size: 11pt;
  }
  .landing-page-widget {
    width: 80%;
    height: 80%;
  }
  .st-widget {
    width: 100%;
    height: 100%;
  }
  .form-wrapper {
    width: 95%;
    padding: 20px;
  }
  .form-title {
    font-size: 12pt;
  }
  .input {
    margin-bottom: 15px;
  }
  .submit-btn {
    padding: 10px 20px;
  }
  /* End of content */
  /* Dashboard */
  .dashboard {
    background-color: rgba(7, 41, 66, 0.8);
    height: 100%;
  }
  .dashboard-content {
    padding: 20px;
    width: 100%;
    height: 100%;
  }
  .search-wrapper {
    margin-bottom: 20px;
  }
  .search-form {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--pureWhite);
    border: 1px solid var(--customYellow);
    border-radius: 15px;
  }
  .search-box {
    background-color: transparent;
    border: none;
    height: 34px;
    width: 80%;
    font-size: 10pt;
    color: var(--customBlue);
    padding: 10px;
  }
  .search-submit {
    padding: 10px;
    height: 44px;
    height: 100%;
    background-color: var(--customYellow);
    border: 1px solid var(--customYellow);
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
    border-width: thick;
  }
  .result-wrapper {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    color: var(--pureWhite);
  }
  .result-title {
    font-size: 16pt;
    text-align: center;
    margin-bottom: 10px;
  }
  .result-date {
    font-size: 9pt;
  }
  .forecast-btn {
    font-size: 10pt;
  }
  .w-icon {
    width: 80px;
    height: 70px;
  }
  .mb20 {
    margin-bottom: 20px;
  }
  .result-details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    font-size: 10pt;
    width: 80%;
  }
  .result-current {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .result-current-desc {
    font-size: 10pt;
  }
  .results-table {
    width: 100%;
    height: 100%;
    border-top: 1px solid var(--pureWhite);
    border-bottom: 1px solid var(--pureWhite);
  }
  .label {
    border-right: 1px solid #FFFFFF;
  }
  .value {
    padding-left: 10px;
  }
}

@media (min-width: 576px) and (max-width: 767.98px) {
  /* Navbar */
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .nav-links {
    flex-direction: row;
  }
  .brand {
    font-size: 14pt;
  }
  .nav-links > li > .link {
    margin-right: 30px;
    color: var(--customBlue);
  }
  .nav-links > li > a {
    font-size: 10pt;
  }
  .small-btn {
    padding: 10px;
    background-color: var(--customBlue);
    border-radius: 10px;
    color: var(--pureWhite);
  }
  /* End of navbar */
  /* Content */
  .landing-page-container {
    width: 100%;
    bottom: 40px;
    position: absolute;
  }
  .landing-page-content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 250px;
  }
  .landing-page-text {
    width: 40%;
    height: 100%;
    margin-right: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .landing-page-title {
    color: var(--pureWhite);
    text-align: center;
    font-size: 14pt;
  }
  .cta-btn {
    padding: 8px 18px;
    background-color: var(--customYellow);
    color: var(--customBlue);
    border: 1px solid var(--customYellow);
    border-radius: 10px;
    margin-top: 20px;
    text-decoration: none;
    font-size: 10pt;
  }
  .landing-page-widget {
    width: 50%;
    margin-left: 10px;
    height: 100%;
  }
  .st-widget {
    width: 100%;
    height: 100%;
  }
  .form-wrapper {
    width: 60%;
    padding: 20px;
  }
  .form-title {
    font-size: 12pt;
  }
  .input {
    margin-bottom: 15px;
  }
  .submit-btn {
    padding: 10px 20px;
  }
  /* End of content */
  /* Dashboard */
  .dashboard {
    background-color: rgba(7, 41, 66, 0.8);
    height: 100%;
  }
  .dashboard-content {
    padding: 20px;
    width: 60%;
    height: 100%;
  }
  .search-wrapper {
    margin-bottom: 20px;
  }
  .search-form {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--pureWhite);
    border: 1px solid var(--customYellow);
    border-radius: 15px;
  }
  .search-box {
    background-color: transparent;
    border: none;
    height: 44px;
    width: 80%;
    font-size: 10pt;
    color: var(--customBlue);
    padding: 10px;
  }
  .search-submit {
    padding: 10px;
    height: 44px;
    height: 100%;
    background-color: var(--customYellow);
    border: 1px solid var(--customYellow);
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
    border-width: thick;
  }
  .result-wrapper {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    color: var(--pureWhite);
  }
  .result-title {
    font-size: 16pt;
    text-align: center;
    margin-bottom: 10px;
  }
  .result-date {
    font-size: 9pt;
  }
  .forecast-btn {
    font-size: 10pt;
  }
  .w-icon {
    width: 80px;
    height: 70px;
  }
  .mb20 {
    margin-bottom: 20px;
  }
  .result-details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    font-size: 10pt;
    width: 80%;
  }
  .result-current {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .result-current-desc {
    font-size: 10pt;
  }
  .results-table {
    width: 100%;
    height: 100%;
    border-top: 1px solid var(--pureWhite);
    border-bottom: 1px solid var(--pureWhite);
  }
  .label {
    border-right: 1px solid #FFFFFF;
  }
  .value {
    padding-left: 10px;
  }
}

@media (min-width: 768px) and (max-width: 991.98px) {
  /* Navbar */
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .nav-links {
    flex-direction: row;
  }
  .brand {
    font-size: 16pt;
  }
  .nav-links > li > .link {
    margin-right: 30px;
    color: var(--customBlue);
  }
  .nav-links > li > a {
    font-size: 11pt;
  }
  .small-btn {
    padding: 10px;
    background-color: var(--customBlue);
    border-radius: 10px;
    color: var(--pureWhite);
  }
  /* End of navbar */
  /* Content */
  .landing-page-container {
    width: 100%;
    bottom: 40px;
    position: absolute;
  }
  .landing-page-content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 250px;
  }
  .landing-page-text {
    width: 30%;
    height: 100%;
    margin-right: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .landing-page-title {
    color: var(--pureWhite);
    text-align: center;
    font-size: 18pt;
  }
  .cta-btn {
    padding: 10px 20px;
    background-color: var(--customYellow);
    color: var(--customBlue);
    border: 1px solid var(--customYellow);
    border-radius: 10px;
    margin-top: 20px;
    text-decoration: none;
    font-size: 11pt;
  }
  .landing-page-widget {
    width: 40%;
    margin-left: 20px;
    height: 100%;
  }
  .st-widget {
    width: 100%;
    height: 100%;
  }
  .form-wrapper {
    width: 80%;
    padding: 20px 40px;
  }
  .form-title {
    font-size: 14pt;
  }
  .input {
    margin-bottom: 20px;
  }
  .submit-btn {
    padding: 10px 20px;
  }
  /* End of content */
  /* Dashboard */
  .dashboard {
    background-color: rgba(7, 41, 66, 0.8);
    height: 100%;
  }
  .dashboard-content {
    padding: 20px;
    width: 60%;
    height: 100%;
  }
  .search-wrapper {
    margin-bottom: 20px;
  }
  .search-form {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--pureWhite);
    border: 1px solid var(--customYellow);
    border-radius: 15px;
  }
  .search-box {
    background-color: transparent;
    border: none;
    height: 44px;
    width: 80%;
    font-size: 10pt;
    color: var(--customBlue);
    padding: 10px;
  }
  .search-submit {
    padding: 10px;
    height: 44px;
    height: 100%;
    background-color: var(--customYellow);
    border: 1px solid var(--customYellow);
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
    border-width: thick;
  }
  .result-wrapper {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    color: var(--pureWhite);
  }
  .result-title {
    font-size: 16pt;
    text-align: center;
    margin-bottom: 10px;
  }
  .result-date {
    font-size: 9pt;
  }
  .forecast-btn {
    font-size: 10pt;
  }
  .w-icon {
    width: 80px;
    height: 70px;
  }
  .mb20 {
    margin-bottom: 20px;
  }
  .result-details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    font-size: 10pt;
    width: 80%;
  }
  .result-current {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .result-current-desc {
    font-size: 10pt;
  }
  .results-table {
    width: 100%;
    height: 100%;
    border-top: 1px solid var(--pureWhite);
    border-bottom: 1px solid var(--pureWhite);
  }
  .label {
    border-right: 1px solid #FFFFFF;
  }
  .value {
    padding-left: 10px;
  }
}

@media (min-width: 992px) and (max-width: 1199.98px) {
  /* Navbar */
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .nav-links {
    flex-direction: row;
  }
  .brand {
    font-size: 18pt;
  }
  .nav-links > li > .link {
    margin-right: 30px;
    color: var(--customBlue);
  }
  .nav-links > li > a {
    font-size: 11pt;
  }
  .small-btn {
    padding: 10px;
    background-color: var(--customBlue);
    border-radius: 10px;
    color: var(--pureWhite);
  }
  /* End of navbar */
  /* Content */
  .landing-page-container {
    width: 100%;
    bottom: 40px;
    position: absolute;
  }
  .landing-page-content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 250px;
  }
  .landing-page-text {
    width: 30%;
    height: 100%;
    margin-right: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .landing-page-title {
    color: var(--pureWhite);
    text-align: center;
    font-size: 24pt;
  }
  .cta-btn {
    padding: 10px 20px;
    background-color: var(--customYellow);
    color: var(--customBlue);
    border: 1px solid var(--customYellow);
    border-radius: 10px;
    margin-top: 20px;
    text-decoration: none;
  }
  .landing-page-widget {
    width: 40%;
    margin-left: 20px;
    height: 100%;
  }
  .st-widget {
    width: 100%;
    height: 100%;
  }
  .form-wrapper {
    width: 50%;
    padding: 30px 40px;
  }
  .form-title {
    font-size: 14pt;
  }
  .input {
    margin-bottom: 20px;
  }
  .submit-btn {
    padding: 10px 20px;
  }
  /* End of content */
  /* Dashboard */
  .dashboard {
    background-color: rgba(7, 41, 66, 0.8);
    height: 100%;
  }
  .dashboard-content {
    padding: 20px;
    width: 40%;
    height: 100%;
  }
  .search-wrapper {
    margin-bottom: 20px;
  }
  .search-form {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--pureWhite);
    border: 1px solid var(--customYellow);
    border-radius: 15px;
  }
  .search-box {
    background-color: transparent;
    border: none;
    height: 44px;
    width: 80%;
    font-size: 10pt;
    color: var(--customBlue);
    padding: 10px;
  }
  .search-submit {
    padding: 10px;
    height: 44px;
    height: 100%;
    background-color: var(--customYellow);
    border: 1px solid var(--customYellow);
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
    border-width: thick;
  }
  .result-wrapper {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    color: var(--pureWhite);
  }
  .result-title {
    font-size: 16pt;
    text-align: center;
    margin-bottom: 10px;
  }
  .result-date {
    font-size: 9pt;
  }
  .forecast-btn {
    font-size: 10pt;
  }
  .w-icon {
    width: 80px;
    height: 70px;
  }
  .mb20 {
    margin-bottom: 20px;
  }
  .result-details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    font-size: 10pt;
    width: 80%;
  }
  .result-current {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .result-current-desc {
    font-size: 10pt;
  }
  .results-table {
    width: 100%;
    height: 100%;
    border-top: 1px solid var(--pureWhite);
    border-bottom: 1px solid var(--pureWhite);
  }
  .label {
    border-right: 1px solid #FFFFFF;
  }
  .value {
    padding-left: 10px;
  }
}

@media (min-width: 1200px) {
  /* Navbar */
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  .nav-links {
    flex-direction: row;
  }
  .brand {
    font-size: 18pt;
  }
  .nav-links > li > .link {
    margin-right: 30px;
    color: var(--customBlue);
  }
  .nav-links > li > a {
    font-size: 11pt;
  }
  .small-btn {
    padding: 10px;
    background-color: var(--customBlue);
    border-radius: 10px;
    color: var(--pureWhite);
  }
  /* End of navbar */
  /* Content */
  .landing-page-container {
    width: 100%;
    bottom: 40px;
    position: absolute;
  }
  .landing-page-content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 250px;
  }
  .landing-page-text {
    width: 30%;
    height: 100%;
    margin-right: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .landing-page-title {
    color: var(--pureWhite);
    text-align: center;
    font-size: 24pt;
  }
  .cta-btn {
    padding: 10px 20px;
    background-color: var(--customYellow);
    color: var(--customBlue);
    border: 1px solid var(--customYellow);
    border-radius: 10px;
    margin-top: 20px;
    text-decoration: none;
  }
  .landing-page-widget {
    width: 40%;
    margin-left: 20px;
    height: 100%;
  }
  .st-widget {
    width: 100%;
    height: 100%;
  }
  .form-wrapper {
    width: 30%;
    padding: 30px 40px;
  }
  .form-title {
    font-size: 14pt;
  }
  .input {
    margin-bottom: 20px;
  }
  .submit-btn {
    padding: 10px 20px;
  }
  /* End of content */
  /* Dashboard */
  .dashboard {
    background-color: rgba(7, 41, 66, 0.8);
    height: 100%;
  }
  .dashboard-content {
    padding: 20px;
    width: 30%;
    height: 100%;
  }
  .search-wrapper {
    margin-bottom: 20px;
  }
  .search-form {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--pureWhite);
    border: 1px solid var(--customYellow);
    border-radius: 15px;
  }
  .search-box {
    background-color: transparent;
    border: none;
    height: 44px;
    width: 80%;
    font-size: 10pt;
    color: var(--customBlue);
    padding: 10px;
  }
  .search-submit {
    padding: 10px;
    height: 44px;
    height: 100%;
    background-color: var(--customYellow);
    border: 1px solid var(--customYellow);
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
    border-width: thick;
  }
  .result-wrapper {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    color: var(--pureWhite);
  }
  .result-title {
    font-size: 16pt;
    text-align: center;
    margin-bottom: 10px;
  }
  .result-date {
    font-size: 9pt;
  }
  .forecast-btn {
    font-size: 10pt;
  }
  .w-icon {
    width: 80px;
    height: 70px;
  }
  .mb20 {
    margin-bottom: 20px;
  }
  .result-details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    font-size: 10pt;
    width: 80%;
  }
  .result-current {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .result-current-desc {
    font-size: 10pt;
  }
  .results-table {
    width: 100%;
    height: 100%;
    border-top: 1px solid var(--pureWhite);
    border-bottom: 1px solid var(--pureWhite);
  }
  .label {
    border-right: 1px solid #FFFFFF;
  }
  .value {
    padding-left: 10px;
  }
}
