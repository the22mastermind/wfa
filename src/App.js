import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import './App.css';
import Header from './components/public/Header';
import LandingPage from './components/public/LandingPage';

function App() {
  return (
    <Router>
      <React.Fragment>
        <Header />
        <Switch>
          <Route path="/">
            <LandingPage />
          </Route>
        </Switch>
      </React.Fragment>
    </Router>
  );
}

export default App;
