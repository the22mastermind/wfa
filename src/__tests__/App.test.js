import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';


it('matches the main App snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
