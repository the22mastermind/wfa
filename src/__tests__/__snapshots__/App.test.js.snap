// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the main App snapshot 1`] = `
Array [
  <nav>
    <a
      className="brand"
      href="/"
    >
      Weather Forecast App
    </a>
    <ul
      className="nav-links"
    >
      <li>
        <a
          className="link"
          href="/signup"
        >
          Sign Up
        </a>
      </li>
      <li>
        <a
          className="small-btn"
          href="/signin"
        >
          Sign In
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="landing-page-container"
  >
    <div
      className="landing-page-content"
    >
      <div
        className="landing-page-text"
      >
        <div
          className="landing-page-title"
        >
          Get real-time weather conditions and forecasts in your city
        </div>
        <a
          className="cta-btn"
          href="/signup"
        >
          Get started
        </a>
      </div>
      <div
        className="landing-page-widget"
      >
        <img
          alt="Weather widget"
          className="st-widget"
          src="widget.svg"
        />
      </div>
    </div>
  </div>,
]
`;
